Use CMake's Build Capabilities

ax3l opened the pull request:
Avoid calling make directly.

Use CMake's parallel build control (CMake 3.12+).

Close #2

ax3l committed:
Use CMake's Build Capabilities

Avoid calling make directly.

ax3l commented on cmake-easyinstall:
```diff
@@ -37,7 +37,7 @@ git clone \
 cd ${tmp_dir}/build
 ${CEI_CMAKE} ../src "${@:2}"
-${CEI_MAKE} -j${CEI_PARALLEL} install
+cmake --build . --target install --config=${CEI_CONFIG}
```
adding `-j ${CEI_PARALLEL}` would require [CMake 3.12+](https://cmake.org/cmake/help/v3.12/manual/cmake.1.html#build-tool-mode)

Flamefire commented on cmake-easyinstall:
```diff
@@ -17,7 +17,7 @@ fi
 # options (undocumented)
 CEI_CMAKE=${CEI_CMAKE:-"cmake"}
-CEI_MAKE=${CEI_MAKE:-"make"}
+CEI_CONFIG=${CEI_CONFIG:-"Release"}
```
You should add that to the configure call `-DCMAKE_BUILD_TYPE=...`

ax3l replied:
Good point. Also, I like `RelWithDebInfo` as default config, too.
(And Spack does so as well.)

ax3l committed:
CMake: Build Parallel

Supported in CMake 3.12.0+
https://cmake.org/cmake/help/v3.12/manual/cmake.1.html#build-tool-mode
